ludwig/schema/hyperopt/search_algorithm.py

```diff
@@ -54,3 +81,675 @@ def _jsonschema_type_mapping(self):
+        description=(
+            "Initial parameter suggestions to be run first. This is for when you already have some good parameters "
+            "you want to run first to help the algorithm make better suggestions for future parameters. Needs to be "
+            "a list of dicts containing the configurations."
+        )
+    )
+
+    max_concurrent: int = schema_utils.NonNegativeInteger(
+        default=0, description="Maximum number of concurrently running trials. If 0 (default), no maximum is enforced."
+    )
+
+    constant_grid_search: bool = schema_utils.Boolean(
+        default=False,
+        description=(
+            "If this is set to True, Ray Tune will first try to sample random values and keep them constant over grid "
+            "search parameters. If this is set to False (default), Ray Tune will sample new random parameters in each "
+            "grid search condition."
+        ),
+    )
+
+    random_state: int = schema_utils.Integer(
+        default=None,
+        allow_none=True,
+        description=(
+            "Seed or numpy random generator to use for reproducible results. If None (default), will use the global "
+            "numpy random generator (np.random). Please note that full reproducibility cannot be guaranteed in a "
+            "distributed environment."
+        ),
+    )
+
+
+@DeveloperAPI
+@register_search_algorithm("ax", dependencies=["ax", "sqlalchemy"])
+@ludwig_dataclass
+class AxSAConfig(BaseSearchAlgorithmConfig):
+    type: str = schema_utils.ProtectedString("ax")
+
+    space: Optional[List[Dict]] = schema_utils.DictList(
+        description=(
+            r"Parameters in the experiment search space. Required elements in the dictionaries are: \“name\” (name of "
+            r"this parameter, string), \“type\” (type of the parameter: \“range\”, \“fixed\”, or \“choice\”, string), "
+            r"\“bounds\” for range parameters (list of two values, lower bound first), \“values\” for choice "
+            r"parameters (list of values), and \“value\” for fixed parameters (single value)."
+        )
+    )
+
+    metric: Optional[str] = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description=(
+            "Name of the metric used as objective in this experiment. This metric must be present in `raw_data` "
+            "argument to `log_data`. This metric must also be present in the dict reported/returned by the Trainable. "
+            "If `None` but a mode was passed, the `ray.tune.result.DEFAULT_METRIC` will be used per default."
+        ),
+    )
+
+    mode: Optional[str] = schema_utils.StringOptions(
+        options=["min", "max", None],
+        default=None,
+        allow_none=True,
+        description=(
+            "One of `{min, max}`. Determines whether objective is minimizing or maximizing the metric attribute. "
+            r"Defaults to \“max\”."
+        ),
+    )
```

Comment:
nit: To simplify the schema and condense it, you can define these base fields either in the Base Class, or as separate methods like you did with `quantization` in the schedulers and then call these here instead of redefining them multiple times (both mode and metric)

Reply:
Maybe even `points_to_evaluate`?

Reply:
I like the latter idea because the search algs don't have consistent configs.